.github/scripts/validate-deploy.sh

```diff
@@ -50,21 +53,77 @@ else
   sleep 30
 fi
 
-DEPLOYMENT="${COMPONENT_NAME}-${BRANCH}"
+SUBSCRIPTION="subscription/ibm-cp-security-operator"
 count=0
-until kubectl get deployment "${DEPLOYMENT}" -n "${NAMESPACE}" || [[ $count -eq 20 ]]; do
-  echo "Waiting for deployment/${DEPLOYMENT} in ${NAMESPACE}"
+until kubectl get "${SUBSCRIPTION}" -n "${NAMESPACE}" || [[ $count -eq 40 ]]; do
+  echo "Waiting for ${SUBSCRIPTION} in ${NAMESPACE}"
   count=$((count + 1))
   sleep 15
 done
 
-if [[ $count -eq 20 ]]; then
-  echo "Timed out waiting for deployment/${DEPLOYMENT} in ${NAMESPACE}"
-  kubectl get all -n "${NAMESPACE}"
+if [[ $count -eq 40 ]]; then
+  echo "Timed out waiting for ${SUBSCRIPTION} in ${NAMESPACE}"
+  kubectl get subscription -n "${NAMESPACE}"
+  exit 1
+fi
+
+CSV="ibm-cp-security-operator"
+count=0
+until kubectl get csv -n "${NAMESPACE}" -l "operators.coreos.com/${CSV}.${NAMESPACE}" || [[ $count -eq 40 ]]; do
+  echo "Waiting for csv ${CSV} in ${NAMESPACE}"
+  count=$((count + 1))
+  sleep 15
+done
+
+if [[ $count -eq 40 ]]; then
+  echo "Timed out waiting for ${CSV} in ${NAMESPACE}"
+  kubectl get csv -n "${NAMESPACE}"
+  exit 1
+else
+  echo "Found CSV"
+  kubectl get csv -n "${NAMESPACE}" -l "operators.coreos.com/${CSV}.${NAMESPACE}"
+fi
+
+## ***** Instance
+
+BRANCH="main"
+SERVER_NAME="default"
+TYPE="instances"
+COMPONENT_NAME="ibm-cp4s-threatmgmt-instance"
+
+if [[ ! -f "argocd/2-services/cluster/${SERVER_NAME}/${TYPE}/${NAMESPACE}-${COMPONENT_NAME}.yaml" ]]; then
+  echo "ArgoCD config missing - argocd/2-services/cluster/${SERVER_NAME}/${TYPE}/${NAMESPACE}-${COMPONENT_NAME}.yaml"
+  exit 1
+fi
+
+echo "Printing argocd/2-services/cluster/${SERVER_NAME}/${TYPE}/${NAMESPACE}-${COMPONENT_NAME}.yaml"
+cat "argocd/2-services/cluster/${SERVER_NAME}/${TYPE}/${NAMESPACE}-${COMPONENT_NAME}.yaml"
+
+if [[ ! -f "payload/2-services/namespace/${NAMESPACE}/${COMPONENT_NAME}/values.yaml" ]]; then
+  echo "Application values not found - payload/2-services/namespace/${NAMESPACE}/${COMPONENT_NAME}/values.yaml"
   exit 1
 fi
 
-kubectl rollout status "deployment/${DEPLOYMENT}" -n "${NAMESPACE}" || exit 1
+echo "Printing payload/2-services/namespace/${NAMESPACE}/${COMPONENT_NAME}/values.yaml"
+cat "payload/2-services/namespace/${NAMESPACE}/${COMPONENT_NAME}/values.yaml"
+
+INSTANCE_NAME="threatmgmt"
+CR="CP4SThreatManagement/${INSTANCE_NAME}"
+
+count=0
+until kubectl get "${CR}" -n "${NAMESPACE}" || [[ $count -eq 30 ]]; do
```

Comment:
Pls check the Readiness of the CustomResourceDefinition by checking for "Ready" State

Reply:
I just verified in a techzone cluster and noticed and it is consuming a number of huge size PVCs, and there is no way to control these by mentioning smaller sizes (though I tried it din't work). Hence not planning to implement that check.